subsys/bluetooth/controller/Kconfig

```diff
@@ -363,6 +349,20 @@ config BT_CTLR_CONN_RSSI
 
 endif # BT_CONN
 
+config BT_CTLR_PHY_2M
```

Comment:
Although this should somehow "work", I personally find the dependencies here a bit complex:

- I understand that the stack should enable 2M support in the controller, if the LE stack supports either the PHY update procedure or the Advertising Extensions. That logic seems fine.

- Then, why do we need to have the dependency on SoC capabilities here? AFAICS, BT_CTRL_PHY already depends on BT_CTLR_PHY_UPDATE_SUPPORT, which is selected when the SoC is capable of doing 2M PHY. I think we should have a single place in the dependency tree where we enforce the SoC capabilities (to allow the 2M or not); either in the _SUPPORT switches, or, here, directly at the controller option.

- This becomes complicated, since BT_CTLR_ADV_EXT depends on BT_CTRL_ADV_EXT_SUPPORT, which is, however, unconditionally selected in the LE stack for Nordic (no SoC dependencies here)

- The expression also looks different from what we have for LE Coded: in this case there is a SoC Kconfig (HAS_HW_NRF_BLE_CODED) that reflects the SoC capabilities. I'd ask for some uniformity here 👍 If you could you introduce a HAS_HW_NRF_RADIO_BLE_2M as well? This would at least make things similar in these 2 expressions.

Reply:
@ioannisg I added HAS_HW_NRF_RADIO_BLE_2M (and similar for Data length requirement). But the complexity here actually comes from the nrf51 data length and 2M phy support without encryption support (BT_CTLR_DATA_LENGTH_CLEAR and BT_CTLR_PHY_2M_NRF).

Atleast that complexity has been moved to a more correct place, with the LLL_VENDOR_NORDIC dealing with this instead of the general feature.
But I think it might have been just as well with:
```
	select BT_CTLR_PHY_2M_SUPPORT if !SOC_SERIES_NRF51X || \
					 BT_CTLR_PHY_2M_NRF
```

@rugeGerritsen Just an FYI that this will now have to be updated for the SD controller hci driver as well.

Reply:
Thanks! I've been thinking about adding the same configurations. Now it looks more maintainable.